[Runtime] Remove RuntimeFunctionCounters in no-assert builds.

mikeash opened the pull request:
A build option to include or exclude RuntimeFunctionCounters. By default, it's enabled when assertions are enabled.

This is a less slash-and-burn approach than #13974 was.

I hope I got the build system bits correct! I would appreciate extra attention on that. Not that there is much else to this.

rdar://problem/35864525

mikeash committed:
[Runtime] Remove RuntimeFunctionCounters in no-assert builds.

A build option to include or exclude RuntimeFunctionCounters. By default, it's enabled when assertions are enabled.

rdar://problem/35864525

Commit:
Update RuntimeInvocationsTracking.h

gparker42 commented:
Isn't there a test of these counters somewhere? Does it need to be updated to prevent failure on no-assert builds?

mikeash commented:
You’re right, the tests need to be disabled too. I’ll fix that and update the PR.

mikeash committed:
[Runtime] Disable RuntimeFunctionCounters tests if the counters are disabled.

Also update how the variable is managed in the build system to allow the test to be conditional based on it, and make it more natural to set it on the command line.

mikeash commented:
@swift-ci please test
